packages/twenty-front/src/modules/object-record/record-board/components/RecordBoard.tsx

```diff
@@ -40,17 +44,25 @@ const StyledBoardHeader = styled.div`
 `;
 
 export const RecordBoard = ({ recordBoardId }: RecordBoardProps) => {
-  const { updateOneRecord, objectMetadataItem } =
+  const { updateOneRecord, selectFieldMetadataItem } =
     useContext(RecordBoardContext);
   const boardRef = useRef<HTMLDivElement>(null);
 
   const { getColumnIdsState, columnsFamilySelector } =
     useRecordBoardStates(recordBoardId);
+
   const columnIds = useRecoilValue(getColumnIdsState());
 
-  const selectFieldMetadataItem = objectMetadataItem.fields.find(
-    (field) => field.type === FieldMetadataType.Select,
-  );
+  const { resetRecordSelection, setRecordAsSelected } =
+    useRecordBoardSelection(recordBoardId);
+
+  useListenClickOutsideByClassName({
+    classNames: ['record-board-card'],
+    excludeClassNames: ['action-bar', 'context-menu'],
+    callback: resetRecordSelection,
+  });
+
+  useScopedHotkeys([Key.Escape], resetRecordSelection, TableHotkeyScope.Table);
```

Comment:
Since both hooks are effects, shouldn't those go into a separated component RecordBoardEffects?

Reply:
As they don't trigger re-render, it's fine to keep them here :)
resetRecordSelection is a recoilCallback